src/Lightning/Node/Api/Invoice.hs

```diff
@@ -27,13 +28,21 @@ data InvoiceReq = InvoiceReq
   }
   deriving (Generic, Show)
 
+data InvoiceLabel = InvoiceLabel {ilLabel :: Maybe Text} deriving (Generic, Show)
```

Comment:
An invoice label is not a `Maybe` though. However, it might still make sense to have a separate type for `InvoiceLabel`, which will be just `Text`, but it should be a `newtype`, not `data`.

Reply:
If there is now attached `{label}` it will just return all invoices. How to describe the possibility of empty label without `Maybe`?